packages/data/addon/services/navi-facts.ts

```diff
@@ -66,34 +65,31 @@ export default class NaviFactsService extends Service {
   }
 
   /**
-   * @method getURL - Uses the adapter to get the download query url for the request
-   * @param {Object} request - request object
-   * @param {Object} [options] - options object
-   * @returns {String} - url for the request
+   * Uses the adapter to get the download query url for the request
+   * @param request - request object
+   * @param options - options object
+   * @returns - url for the request
    */
-  getDownloadURL(request: RequestV2, options: RequestOptions) {
+  @task *getDownloadURL(request: RequestV2, options: RequestOptions): TaskGenerator<string> {
     const { type: dataSourceType } = getDataSource(request.dataSource);
     const adapter = this._adapterFor(dataSourceType);
-    return adapter.urlForDownloadQuery(request, options);
+    return yield adapter.urlForDownloadQuery(request, options);
```

Comment:
do we need to call `perform` here

Reply:
Good catch 👍 , unfortunately the typescript for calling is not very helpful